docs/config/shared-options.md

```diff
@@ -135,7 +136,9 @@ A package with conditional exports may have the following `exports` field in its
 
 Here, `import` and `require` are "conditions". Conditions can be nested and should be specified from most specific to least specific.
 
-Vite has a list of "allowed conditions" and will match the first condition that is in the allowed list. The default allowed conditions are: `import`, `module`, `browser`, `default`, and `production/development` based on current mode. The `resolve.conditions` config option allows specifying additional allowed conditions.
+Some of the default conditions (`production`, `development`) are only applied when the requirements are met. For example, `production` is only applied when `process.env.NODE_ENV === 'production'`. The `resolve.conditions` config option allows specifying additional allowed conditions and those conditions will be applied unconditionally.
+
+Note that `import`, `require`, `default` conditions are always applied if the requirements are met.
```

Comment:
What is a good use case for not passing `production` and `development`?

I'm wondering if we should treat these two specially, like you do with `import`, `require`, `default`. Given that they are the only ones that are conditionally applied, it feels cleaner for `resolve.conditions` to be "the conditions that are always applied". We could add a way to pass a function for each condition to specify 
the conditions for example if we want to explicitly define them, or ask for 
`['module', 'browser', 'node', process.env.NODE_ENV]`, maybe this should be the default if we keep it in the conditions?)

Reply:
As we discussed in the team meeting, I introduced `DEV_PROD_CONDITION` variable that will be replaced with `production` or `development`: https://github.com/vitejs/vite/pull/18395/commits/49850110d395269608df4c718f5c5eb82d72b47f